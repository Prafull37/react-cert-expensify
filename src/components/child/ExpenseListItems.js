import React from 'react';
import {Link} from 'react-router-dom';


const ExpenseListItems=(props)=>(<div>
   <Link to={`/edit/${props.id}`}>
        <h6>{props.description}</h6>
        </Link>
        
        <div>{props.amount} - {props.createdAt}</div>
        
    </div>
)

export default ExpenseListItems;