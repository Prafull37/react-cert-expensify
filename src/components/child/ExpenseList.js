import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from "../../utils/sortAndFilter"
import ExpenseListItems from './ExpenseListItems';

export const ExpenseList=(props)=>{
    const {expenses}=props;
  
    return (
        <div>

           <h3>Expense List</h3>
            {expenses.length===0?<div>No Expenses</div>:
            <div>{expenses.map((expense)=><ExpenseListItems key={expense.id} {...expense}/>)}</div>
            }
        </div>
    )
}

const mapStateToProps=(state)=>{
  console.log("state",state)
    return {

       expenses:selectExpenses(state.expenses,state.filter)
   }
}

export default connect(mapStateToProps)(ExpenseList);