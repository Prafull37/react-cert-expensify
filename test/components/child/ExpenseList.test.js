import React from "react"
import {shallow} from "enzyme";
import {ExpenseList} from "../../../src/components/child/ExpenseList";
import expenses from "../../fixtures/expense";

test("should render the ExpenseList with expenses",()=>{
    const wrapper=shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
})

test("should render the ExpenseList with []",()=>{
    const wrapper=shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
})