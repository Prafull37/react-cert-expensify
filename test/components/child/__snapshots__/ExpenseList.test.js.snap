// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpenseList with [] 1`] = `
<div>
  <h3>
    Expense List
  </h3>
  <div>
    No Expenses
  </div>
</div>
`;

exports[`should render the ExpenseList with expenses 1`] = `
<div>
  <h3>
    Expense List
  </h3>
  <div>
    <ExpenseListItems
      amount={195}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      notes=""
    />
    <ExpenseListItems
      amount={109500}
      createdAt={-345600000}
      description="Rent"
      id="2"
      key="2"
      notes=""
    />
    <ExpenseListItems
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      notes=""
    />
  </div>
</div>
`;
